sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -3191,6 +3193,16 @@ private void mergeCase(CaseDataDto leadCaseData, CaseDataDto otherCaseData, bool
 
 			documentService.ensurePersisted(document);
 		}
+
+		// 7 Exposures - Make sure there are no two probable infection environments
+		// if there are more than 2 exposures marked as probable infection environment, find the one that originates from the otherCase and set it to false
+		// the one originating from the otherCase should always be found at the higher index
+		List<Exposure> probableExposuresList =
+			leadCase.getEpiData().getExposures().stream().filter(Exposure::isProbableInfectionEnvironment).collect(Collectors.toList());
+		if (probableExposuresList.size() >= 2) { // should never be > 2, but still make sure
+			probableExposuresList.get(probableExposuresList.size() - 1).setProbableInfectionEnvironment(false);
```

Comment:
I was thinking about this too, but didn't do it because the situation really should not occur. But you are right that it would become relevant when merging more than 2 cases, so I will implement it